lib/scholar/linear/svm.ex

```diff
@@ -0,0 +1,257 @@
+      type = Nx.type(t)
+
+      if Nx.Type.float?(type) and not Nx.Type.complex?(type) do
+        Nx.as_type(t, target_type)
+      else
+        t
+      end
+    end)
+  end
+
+  # SVM training loop
+  defnp fit_n(x, y, coef, bias, coef_optimizer_state, bias_optimizer_state, opts) do
+    iterations = opts[:iterations]
+    num_classes = opts[:num_classes]
+    optimizer_update_fn = opts[:optimizer_update_fn]
+    eps = opts[:eps]
+
+    {{final_coef, final_bias}, _} =
+      while {{coef, bias},
+             {x, iterations, y, coef_optimizer_state, bias_optimizer_state,
+              has_converged = Nx.broadcast(Nx.u8(0), {num_classes}), eps, iter = 0}},
+            iter < iterations and not Nx.all(has_converged) do
+        # ++++ inner while ++++++
+        {{coef, bias, has_converged, coef_optimizer_state, bias_optimizer_state}, _} =
+          while {{coef, bias, has_converged, coef_optimizer_state, bias_optimizer_state},
+                 {x, y, iterations, iter, eps, j = 0}},
+                j < num_classes do
+            y_j = y == j
+            coef_j = Nx.take(coef, j)
+            bias_j = Nx.take(bias, j)
+
+            {loss, {coef_grad, bias_grad}} = loss_and_grad(coef_j, bias_j, x, y_j, opts[:loss_fn])
+            grad = Nx.broadcast(0.0, {num_classes, Nx.axis_size(x, 1)})
+            coef_grad = Nx.put_slice(grad, [j, 0], Nx.new_axis(coef_grad, 0))
+
+            {coef_updates, coef_optimizer_state} =
+              optimizer_update_fn.(coef_grad, coef_optimizer_state, coef)
+
+            coef = Polaris.Updates.apply_updates(coef, coef_updates)
+
+            grad = Nx.broadcast(0.0, {num_classes})
+            bias_grad = Nx.put_slice(grad, [j], Nx.new_axis(bias_grad, 0))
+
+            {bias_updates, bias_optimizer_state} =
+              optimizer_update_fn.(bias_grad, bias_optimizer_state, bias)
+
+            bias = Polaris.Updates.apply_updates(bias, bias_updates)
+
+            has_converged_j = Nx.sum(Nx.abs(loss)) < Nx.axis_size(x, 0) * eps
+
+            has_converged =
+              Nx.indexed_put(
+                has_converged,
+                Nx.new_axis(j, -1),
+                has_converged_j
+              )
+
+            {{coef, bias, has_converged, coef_optimizer_state, bias_optimizer_state},
+             {x, y, iterations, iter, eps, j + 1}}
+          end
+
+        # ++++ end inner while ++++++
+
+        {{coef, bias},
+         {x, iterations, y, coef_optimizer_state, bias_optimizer_state, has_converged, eps,
+          iter + 1}}
+      end
+
+    %__MODULE__{
+      coefficients: final_coef,
+      bias: final_bias
+    }
+  end
+
+  defnp loss_and_grad(coeff, bias, xs, ys, loss_fn) do
+    value_and_grad({coeff, bias}, fn {coeff, bias} ->
+      y_pred = predict(coeff, bias, xs)
+      loss_fn.(y_pred, ys)
+    end)
+  end
+
+  defnp predict(coeff, bias, xs) do
+    # Nx.dot(xs, [1], coeff, [1]) + bias
+    Nx.dot(xs, Nx.transpose(coeff)) + bias
```

Comment:
```suggestion
    Nx.dot(xs, [-1], coeff, [-1]) + bias
```

Why not use the dot/4 trick here?